subprojects/core/src/main/java/org/gradle/cache/internal/DefaultGeneratedGradleJarCache.java

```diff
@@ -20,30 +20,27 @@
 import org.gradle.cache.CacheRepository;
 import org.gradle.cache.FileLockManager;
 import org.gradle.cache.PersistentCache;
-import org.gradle.internal.operations.BuildOperationContext;
-import org.gradle.internal.operations.BuildOperationDescriptor;
-import org.gradle.internal.operations.BuildOperationExecutor;
-import org.gradle.internal.operations.RunnableBuildOperation;
+import org.gradle.internal.classpath.CachedJarFileStore;
 
 import java.io.Closeable;
 import java.io.File;
+import java.util.Collections;
+import java.util.List;
 
 import static org.gradle.cache.internal.filelock.LockOptionsBuilder.mode;
 
-public class DefaultGeneratedGradleJarCache implements GeneratedGradleJarCache, Closeable {
+public class DefaultGeneratedGradleJarCache implements GeneratedGradleJarCache, Closeable, CachedJarFileStore {
```

Comment:
This has been moved up to user home scope so that it can actually be picked up as an immutable location by the other user-home-scoped services. The build operation handling was moved one level higher as that infrastructure is not available in user home scope.